src/redturtle/volto/restapi/services/search/get.py

```diff
@@ -5,31 +5,73 @@
 from plone.restapi.services import Service
 from zope.component import getMultiAdapter
 
+# search for 'ranking' in 'SearchableText' and rank very high
+# when the term is in 'Subject' and high when it is in 'Title'.
+# print the id and the normalized rank
+try:
+    from Products.AdvancedQuery import RankByQueries_Sum
+    from Products.AdvancedQuery import Eq
+    # from Products.AdvancedQuery import In
+
+    HAS_ADVANCEDQUERY = True
+except ImportError:
+    HAS_ADVANCEDQUERY = False
+
+
+# custom search handler
+
 
 class SearchHandler(OriginalHandler):
+    def is_advanced_query(self, query):
+        simple_query = query.get("SimpleQuery", None)
+        if simple_query:
+            return False
+        # TODO: aggiungere un parametro in registry per abilitare/disabilitare
+        if HAS_ADVANCEDQUERY and "SearchableText" in query:
+            return True
+        return False
+
+    # XXX: sarebbe meglio una monkeypatch a catalog.searchResults ? eviterebbe di
+    #      ripotare tutto il codice di search qui
```

Comment:
a parer mio no, è questo il modo giusto. Soprattutto in un pacchetto che deve essere da base per altri progetti.